File: jupyter-js-widgets/src/widget_core.ts
// Copyright (c) Jupyter Development Team.
// Distributed under the terms of the Modified BSD License.

// widget_core implements some common patterns for the core widget collection
// that are not to be used directly by third-party widget authors.

import {
    DOMWidgetModel, WidgetModel
} from './widget';

import {
    DescriptionModel
} from './widget_description';

import * as _ from 'underscore';

export
class CoreWidgetModel extends WidgetModel {
    defaults() {
        return _.extend(super.defaults(), {
            _model_name: 'CoreWidgetModel',
        });
    }
}

export
class CoreDOMWidgetModel extends DOMWidgetModel {
    defaults() {
        return _.extend(super.defaults(), {
            _model_name: 'CoreDOMWidgetModel',
        });
    }
}

export
class CoreDescriptionModel extends DescriptionModel {
    defaults() {
        return _.extend(super.defaults(), {
            _model_name: 'CoreDescriptionModel',
        });
    }
}

File: ipywidgets/widgets/widget_core.py
# Copyright (c) Jupyter Development Team.
# Distributed under the terms of the Modified BSD License.

"""Base widget class for widgets provided in Core"""

from .widget import Widget
from .._version import __frontend_version__

from traitlets import Unicode

class CoreWidget(Widget):

    _model_module_version = Unicode(__frontend_version__).tag(sync=True)
    _view_module_version = Unicode(__frontend_version__).tag(sync=True)

